### ISSUE TITLE
Spawning a child NOB by instantiate uses global position and rotation instead of local

### ISSUE THREAD

**Issue by AhonenAJ** [FirstGearGames/FishNet#660]
**General**
Unity version: 2022.3.23f1
Fish-Networking version: 4.2.2R
Discord link: None (seems like a straightforward bug to me)

**Description**
When you spawn a child NOB from a prefab, the child's initial coordinates are treated as global instead of local, causing it to position incorrectly in relation to its parent.

I'm not sure if there are other problem spots but at least this change fixed it for me in DefaultObjectPool.RetrieveObject.GetFromInstantiate:

```
NetworkObject result = Instantiate(prefab, pos, rot, parent);
result.transform.localScale = scale;
```
->
```
NetworkObject result = Instantiate(prefab, parent);
var tf = result.transform;
tf.localPosition = pos;
tf.localRotation = rot;
tf.localScale = scale;
```

**Replication**
1. Spawn an object at non-zero coordinates in the game world
2. Immediately spawn a child object (with no network transform) for the first spawn at (0, 0, 0) local position

**Expected behavior**
On the client, the child object is spawned at (0, 0, 0) local position under the parent and not at (0, 0, 0) global position.

Note that, with a network transform added to the child object, it worked for me as long as the client was already joined when the spawn triggered. If the spawn had triggered before the client joined, then the child position ended up being wrong again and the network transform did nothing to fix the issue.

**Comment by FirstGearGames** [comment]
That could have other consequences but this does sound like a bug. Thanks for the report.

**Comment by FirstGearGames** [comment]
I reviewed further and the default is to write local properties. Your fix should do the trick! I'll review further but I think we're okay with it.

**Comment by FirstGearGames** [comment]
Tested and working!
```cs
                    prefab.transform.OutLocalPropertyValues(nullableLocalPosition, nullableLocalRotation, nullableLocalScale, out Vector3 pos, out Quaternion rot, out Vector3 scale);                    
                    if (parent != null)
                    {
                        //Convert pos and rot to world values for the instantiate.
                        pos = parent.TransformPoint(pos);
                        rot = (parent.rotation * rot);
                    }
                    NetworkObject result = Instantiate(prefab, pos, rot, parent);
                    result.transform.localScale = scale;
```
Resolved in 4.3.0
